aten/src/ATen/functorch/BatchRulesLinearAlgebra.cpp

```diff
@@ -280,6 +280,62 @@ static void expect_at_least_rank(
       rank, " dimensions instead.");
 }
 
+fourOutputs _scaled_dot_product_efficient_attention_batch_rule(
+  const Tensor& query, optional<int64_t> query_bdim,
+  const Tensor& key, optional<int64_t> key_bdim,
+  const Tensor& value, optional<int64_t> value_bdim,
+  bool compute_log_sumexp,
+  double dropout_p,
+  bool is_causal,
+  c10::optional<double> scale
+) {
+  const auto query_ = query_bdim.has_value() ? reshape_dim_into(*query_bdim, 0, query) : query;
+  const auto key_ = key_bdim.has_value() ? reshape_dim_into(*key_bdim, 0, key) : key;
+  const auto value_ = value_bdim ? reshape_dim_into(*value_bdim, 0, value) : value;
+
+  const auto res = at::_scaled_dot_product_efficient_attention(query_, key_,
```

Comment:
You may be able to make this a bit more readable by using structured bindings in C++17 to extract the values directly from the std::tuple.